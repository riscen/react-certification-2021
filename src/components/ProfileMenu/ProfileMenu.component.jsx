import React from 'react';
import { Menu, MenuList, MenuItem } from '@material-ui/core';

import './ProfileMenu.styles.css';

function ProfileMenu(props) {
  const { isProfileMenuActive, closeMenu } = props;
  const isMenuOpen = Boolean(isProfileMenuActive);

  return (
    <Menu
      id="profile-menu"
      anchorEl={isProfileMenuActive}
      keepMounted
      open={isMenuOpen}
      onClose={() => closeMenu(null)}
    >
      <MenuList autoFocusItem={isMenuOpen}>
        <MenuItem>
          <span>Dark theme</span>
          <>{/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}</>
          <label className="toggle-checkbox">
            <input type="checkbox" />
            <span className="slider" />
          </label>
        </MenuItem>
        <MenuItem>
          <span>Sign out</span>
        </MenuItem>
      </MenuList>
    </Menu>
  );
}

export default ProfileMenu;
