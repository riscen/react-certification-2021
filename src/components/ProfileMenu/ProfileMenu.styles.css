/***************THEME TOGGLE****************/
.toggle-checkbox {
  display: inline-block;
  height: 100%;
  position: relative;
  width: 50px;
}

.toggle-checkbox input {
  height: 0;
  opacity: 0;
  width: 0;
}

.slider {
  background-color: #ccc;
  border-radius: 4px;
  bottom: 0;
  cursor: pointer;
  left: 0;
  margin-left: 10px;
  position: absolute;
  right: 0;
  top: 0;
  transition: 0.4s;
  -webkit-transition: 0.4s;
}

.slider:before {
  background-color: #ffffff;
  bottom: 3px;
  border-radius: 4px;
  content: '';
  height: 1.2em;
  left: 3px;
  position: absolute;
  width: 1em;
  transition: 0.4s;
  -webkit-transition: 0.4s;
}

input:checked + .slider {
  background-color: #191a5e;
}

input:checked + .slider:before {
  transform: translateX(15px);
  -ms-transform: translateX(15px);
  -webkit-transform: translateX(15px);
}
