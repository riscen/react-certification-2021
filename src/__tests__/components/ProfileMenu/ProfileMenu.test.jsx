import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';

import ProfileMenu from '../../../components/ProfileMenu';

describe('ProfileMenu', () => {
  beforeEach(() => {
    configure({ adapter: new Adapter() });
  });

  it('Renders ProfileMenu', () => {
    const profileMenu = mount(
      <ProfileMenu isProfileMenuActive={null} closeMenu={jest.fn()} />
    );
    expect(profileMenu).toMatchSnapshot();
  });

  it('Renders Menu', () => {
    const profileMenu = mount(
      <ProfileMenu isProfileMenuActive={null} closeMenu={jest.fn()} />
    );
    expect(profileMenu.find('#profile-menu')).toBeDefined();
  });

  it('Renders Change Theme Checkbox', () => {
    const profileMenu = mount(
      <ProfileMenu isProfileMenuActive={null} closeMenu={jest.fn()} />
    );
    expect(profileMenu.find('label')).toBeDefined();
    expect(profileMenu.find('input[type="checkbox"]')).toBeDefined();
    expect(profileMenu.find('.slider')).toBeDefined();
  });
});
