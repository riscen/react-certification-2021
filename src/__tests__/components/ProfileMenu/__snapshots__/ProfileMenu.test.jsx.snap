// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileMenu Renders ProfileMenu 1`] = `ReactWrapper {}`;
